extensions/hexpm/src/hooks/use-packages-query.ts

```diff
@@ -0,0 +1,23 @@
+import { useFetch } from "@raycast/utils";
+import { Package } from "../types";
+
+type HexPackagesResponse = Package[];
+
+export function usePackagesQuery(props: {
+  search: string;
+  sort: "name" | "total_downloads" | "recent_downloads" | "inserted_at" | "updated_at";
+  onError?(error: Error): void;
+}) {
+  const url = new URL("https://hex.pm/api/packages");
+  url.searchParams.append("sort", props.sort);
+  url.searchParams.append("search", props.search);
+
+  return useFetch<HexPackagesResponse, HexPackagesResponse, HexPackagesResponse>(url.toString(), {
+    method: "GET",
+    headers: { Accept: "application/json" },
+    execute: props.search.length > 0,
```

Comment:
This is NOT working as intended; it is caching the initial request, and it is not respecting the `execute` value at all!